phoenix-core-client/src/main/java/org/apache/phoenix/jdbc/HighAvailabilityPolicy.java

```diff
@@ -44,63 +46,98 @@ public Connection provide(HighAvailabilityGroup haGroup, Properties info,
             FailoverPhoenixContext context = new FailoverPhoenixContext(info, haGroup, haURLInfo);
             return new FailoverPhoenixConnection(context);
         }
+
         @Override
         void transitClusterRole(HighAvailabilityGroup haGroup, ClusterRoleRecord oldRecord,
```

Comment:
As discussed, maybe good to add the check for both roles being ACTIVE, due to misconfiguration and log it.